### DISCUSSION TITLE
FadeEffect does not impact model layer with point data

### DISCUSSION THREAD

**Discussion by shajder** [pelicanmapping/osgearth#2278] [Support Q&A]
Hello community

Please correct me if I am wrong but it seems for me like `fading` filter does not affect blending of an icon, only text. This is fragment of my .earth file:

```
  <model name="name" driver="feature_geom" max_range="20000" min_range="40">
    <fading duration = "1.0" max_range = "1000"  attenuation_distance = "1000" />
    <styles>
      <style type="text/css">
        city_label {
          icon:           "image.tga";
          icon-declutter: false;
          icon-align:     center-center;
          icon-placement: centroid;
          icon-density:   0.1;
          icon-scale:     1.0;
          text-font:      font.ttf;
          text-size:      40;
          text-content:   getName();
          text-encoding:  utf-8;
          text-align:     left-center;
          text-fill:      #fffffff0;
          text-halo:      #000000FF;
          text-priority:  getTextPriority();
          altitude-clamping: none;
        }
```

Here is possible solution which works for me:

https://github.com/gwaldron/osgearth/blob/f8d2a1341ddb2aa4ed90d6db885cf098ccb1e8df/src/osgEarth/PlaceNode.cpp#L56

`color = texture(oe_PlaceNode_tex, oe_PlaceNode_texcoord) * color;`

Am I missing some proper solution here ?

cheers
Marcin

**Comment by gwaldron** [comment]
I think you are correct, this is just a missing feature. If you can submit a PR we'll review and merge. Thanks.

**Reply by shajder** [reply]
Sorry for confusion but my initial impression I have solved the problem wasn't correct. Indeed I've noticed change in blending of an icons but it does not come from `FadeEffect`. I dive deeper into the problem and I see now the source of the problem lays around `ScreenSpaceLayoutDeclutter` infrastructure which modifies both matrices and vertex attributes. This ruins the logic of `FadeEffect` shaders completely due to the fact that `(-VertexView.z)` used in FadeEffect changes it's meaning in screen-space layout (moreover I see similar code in `VisualLayer` class). Still I haven't found proper solution for this problem.
